tests/FixtureBuilder/Fixture.php

```diff
@@ -0,0 +1,275 @@
+<?php
+
+namespace Tuf\Tests\FixtureBuilder;
+
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * This is our TUF fixture builder.
+ *
+ * This class is MEANT FOR TESTING PURPOSES ONLY and should absolutely not be
+ * used in production. It is meant to generate complete TUF repositories, but
+ * doesn't necessarily support all of the functionality from the Python TUF
+ * tool it was derived from.
+ */
+class Fixture
+{
+    public readonly Root $root;
+
+    public readonly Timestamp $timestamp;
+
+    public readonly Snapshot $snapshot;
+
+    /**
+     * @var \Tuf\Tests\FixtureBuilder\Targets[]
+     */
+    public array $targets = [];
+
+    public readonly string $serverDir;
+
+    private readonly string $clientDir;
+
+    private readonly string $targetsDir;
+
+    public function __construct(
+        ?string $baseDir = null,
+        ?\DateTimeImmutable $expires = null,
+        private readonly Filesystem $fileSystem = new Filesystem(),
+    ) {
+        // By default, create the fixture in a temporary directory.
+        $baseDir ??= uniqid(sys_get_temp_dir() . '/TUF_Fixture_');
+
+        // If there's any previous data in the directory, delete it.
+        $this->serverDir = $baseDir . '/server';
+        $this->clientDir = $baseDir . '/client';
+        $this->targetsDir = $baseDir . '/targets';
+        $fileSystem->remove([
+            $this->serverDir,
+            $this->clientDir,
+            $this->targetsDir,
+        ]);
+
+        // By default, all metadata we generate should expire in a year. This
+        // can be overridden for individual metadata objects.
+        $expires ??= new \DateTimeImmutable('+1 year');
+
+        $this->root = new Root($expires, [
+            Key::fromStaticList(),
+        ]);
+        $this->timestamp = new Timestamp($this->root, $expires, [
+            Key::fromStaticList(),
+        ]);
+
+        $this->snapshot = new Snapshot($this->root, $this->timestamp, $expires, [
+            Key::fromStaticList(),
+        ]);
+        // Disable hashes and file sizes in the snapshot metadata, which matches
+        // the default configuration of the Python TUF tool.
```

Comment:
I think we should state reason we want this to be this way instead of referring to the python tool.